app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -255,10 +200,74 @@ def update(self, experiment, data):
         return experiment
 
 
+class NimbusStatusValidationMixin:
+    def validate(self, data):
+        data = super().validate(data)
+
+        restrictive_statuses = {
+            "status": NimbusConstants.STATUS_CHANGE_ONLY,
+            "publish_status": NimbusConstants.PUBLISH_STATUS_CHANGE_ONLY,
+        }
+
+        if self.instance:
+            status = self.instance.status
+
+            for status_field, statuses in restrictive_statuses.items():
+                instance_value = getattr(self.instance, status_field)
+                if instance_value in statuses:
+                    if set(data.keys()) == {status_field}:
+                        return data
+                    else:
```

Comment:
You can take this whole early return out if you want and just

```py
if not set(.....):
    raise serializers.ValidationError(...)
```

I'm not a huge fan of early returns if they can be avoided, makes it harder to trace the flow of a function.

Reply:
I think the early return is to avoid catching on the following case?

Reply:
Don't you WANT to check that case if the first case is valid?

Reply:
If you only want to check the second case if the first case is valid then i'd write it as a single if/else block instead of the early return to make it clearer?

Reply:
If I'm understanding this correctly, if the first check passes (the status or publish_status permits any field to be changed), we don't care if the status is not in the list of statuses that only allows status updates.

So yes, I could remove the early return, but to avoid hitting that second check I'd also need to set up a variable inside the for-loop that determines whether or not to perform that second check. I think?

Reply:
a la https://github.com/mozilla/experimenter/pull/4887/commits/e3775186f73bfe6c6f7b332a0dd6ea2f9be603a8